internal/monitors/statsd/parser.go

```diff
@@ -0,0 +1,153 @@
+package statsd
+
+import (
+	"strings"
+)
+
+type fieldPattern struct {
+	substrs        []string
+	startWithField bool
+}
+
+type converter struct {
+	pattern    *fieldPattern
+	metric     *fieldPattern
+	dimensions []string
+}
+
+func initConverter(input *converterInput) *converter {
+	if input.Pattern == "" {
+		logger.Error("[pattern] is required for a converter")
+		return nil
+	}
+	if input.MetricName == "" {
+		logger.Error("[metricName] is required for a converter")
+		return nil
+	}
+
+	pattern := parseFields(input.Pattern)
+	metric := parseFields(input.MetricName)
+
+	return &converter{
+		pattern:    pattern,
+		metric:     metric,
+		dimensions: input.Dimensions,
+	}
+}
+
+// parsePattern takes a pattern string and convert it into parsed fieldPattern object
+func parseFields(p string) *fieldPattern {
+	var substrs []string
+	var startWithField bool
+
+	l := len(p)
+	wa := 0
+	var wb, wc int
+
+	for wa < l {
+		wb = wa + strings.Index(p[wa:], "{")
+		wc = wa + strings.Index(p[wa:], "}")
+
+		if wa == 0 && wb == 0 {
+			startWithField = true
+		}
+
+		if wa != wb {
+			// edge case : pattern not ending with { and }
+			if wb == -1 {
+				return nil // Invalid pattern, skip.
```

Comment:
Would probably be helpful for this function to also return an error explaining that there was a problem parsing and then have that logged somewhere so the user gets some feedback.

Reply:
Don't forget about this.

Reply:
Don't know why it's not showing up, but it's been added.
https://github.com/signalfx/signalfx-agent/blob/ae6f1373f7e367934f9015206a2a02208214cecb/internal/monitors/statsd/parser.go#L51